bin/update_cucumber

```diff
@@ -109,27 +117,44 @@ class {
     }
 
     /**
-     * @return list<GitHubReleaseArray>
+     * @phpstan-return non-empty-list<TGitHubReleaseArray>
      */
     private function listGitHubReleases(string $repo): array
     {
         // GitHub requires API requests to send a user_agent header for tracing
         ini_set('user_agent', 'https://github.com/Behat/Gherkin updater');
         $releases = $this->jsonDecodeArray($this->getContents("https://api.github.com/repos/$repo/releases"));
 
+        assert($releases !== []);
+
         // Releases API is in order of github release ID, this may not be version order (e.g. major releases may be
         // interspersed with point releases to older versions). So sort it most recent -> oldest by version number.
-        usort($releases, static fn (array $a, array $b) => version_compare($b['tag_name'], $a['tag_name']));
+        usort($releases, static function (mixed $a, mixed $b) {
```

Comment:
At this point, PHPStan still sees `$release` as an "array of stuff". If we want to avoid this excessive type checking within sorting, maybe we should move sorting after the array is formally defined (e.g. instead if directly returning the next `array_map`, we put it in a variable, sort and then return it).

I'm fine with either approach.

Reply:
Yeah I think if we *have* to check these are all strings (again noting that it would also be fine for this to just blow up if somehow github's schema changed and they gave us something that couldn't go into version_compare) then we should array_map and then sort.

Reply:
OK, I'll go for it.